### PR TITLE
UNIT TESTS

### PR DESCRIPTION
**Pull request by EdwinRozario**

### CODE REVIEW

File: app/controllers/safety_cone_mountable/cones_controller.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
 require_dependency "safety_cone_mountable/application_controller"
+
 module SafetyConeMountable
   class ConesController < ApplicationController
-    http_basic_authenticate_with name: SafetyConeMountable.auth[:username], 
+    if Rails.env.production?
+    http_basic_authenticate_with name: SafetyConeMountable.auth[:username],
```
Comment (hapiben) [inline app/controllers/safety_cone_mountable/cones_controller.rb]: Indent this line.

File: safety_cone_mountable.gemspec [hunk 2]
```diff
@@ -5,20 +5,27 @@ require "safety_cone_mountable/version"
 
 # Describe your gem and declare its dependencies:
 Gem::Specification.new do |s|
```
Comment (hapiben) [inline safety_cone_mountable.gemspec]: Fix the indention for this file.

File: spec/features/edit_page_features_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      config.redis = $redis
+      config.auth = { username: 'admin', password: 'password' }
+    end
+  }
+
+  scenario 'A user sees the safety cone Configuration page header' do
+    @cones = SafetyConeMountable.cones
+    save_and_open_page
+
+    binding.pry
```
Comment (hapiben) [inline spec/features/edit_page_features_spec.rb]: Remove the `binding.pry`

File: spec/features/edit_page_features_spec.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        keys: '1234'
+      )
+      config.redis = $redis
+      config.auth = { username: 'admin', password: 'password' }
+    end
+  }
+
+  scenario 'A user sees the safety cone Configuration page header' do
+    @cones = SafetyConeMountable.cones
+    save_and_open_page
```
Comment (hapiben) [inline spec/features/edit_page_features_spec.rb]: Remove the `save_and_open_page`

File: spec/features/edit_page_features_spec.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        measure: 'notice',
+        redis: $redis,
+        keys: '1234'
+      )
+      config.redis = $redis
+      config.auth = { username: 'admin', password: 'password' }
+    end
+  }
+
+  scenario 'A user sees the safety cone Configuration page header' do
```
Comment (hapiben) [inline spec/features/edit_page_features_spec.rb]: You can simplify this by just saying:

`scenario 'page header text'`

File: spec/features/home_page_features_spec.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+  scenario 'A user sees the safety cone Configuration page header' do
+    @cones = SafetyConeMountable.cones
+    visit root_path
+    within '.s4' do
+      expect(page).to have_text('Safety Cone');
+    end
+  end
+
+  scenario 'A user sees the safety cone Configuration page header' do
```
Comment (hapiben) [inline spec/features/home_page_features_spec.rb]: The scenario text is the same as the one above.

File: spec/features/home_page_features_spec.rb [hunk 7]
```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+require 'spec_helper'
```
Comment (hapiben) [inline spec/features/home_page_features_spec.rb]: Move the other `require` inside`rails_helper.rb`

File: spec/features/edit_page_features_spec.rb [hunk 8]
```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
```
Comment (hapiben) [inline spec/features/edit_page_features_spec.rb]: Move the other `require` inside`rails_helper.rb`

File: spec/support/capybara.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# spec/support/capybara.rb
+require 'capybara/rspec'
+require 'capybara/rails'
+
+Capybara.app = SafetyConeMountable::Engine
+
+RSpec.configure do |config|
```
Comment (hapiben) [inline spec/support/capybara.rb]: Move this as part of `rails_helper.rb`. The file should only show configurations for Capybara.

File: app/controllers/safety_cone_mountable/cones_controller.rb (hunk 1 above)
Comment (EdwinRozario) [inline app/controllers/safety_cone_mountable/cones_controller.rb]: I think this shuould be unless `env.test?`

Comment (hapiben) [inline app/controllers/safety_cone_mountable/cones_controller.rb]: Agree.

**Pull request merged**